pkg/tabulator/tabstate.go

```diff
@@ -245,3 +260,46 @@ func TabStatePath(configPath gcs.Path, tabPrefix, dashboardName, tabName string)
 	}
 	return np, nil
 }
+
+func calculateState(ctx context.Context, client gcs.Client, log *logrus.Entry, fromPath, toPath gcs.Path, confirm bool) error {
+	r, _, err := client.Open(ctx, fromPath)
+	if err != nil {
+		return fmt.Errorf("client.Open(%s): %w", fromPath.String(), err)
+	}
+	defer r.Close()
+	z, err := zlib.NewReader(r)
+	if err != nil {
+		return fmt.Errorf("zlib.NewReader: %w", err)
+	}
+	defer z.Close()
+	buf, err := ioutil.ReadAll(z)
+	if err != nil {
+		return fmt.Errorf("ioutil.ReadAll: %w", err)
+	}
+	var g statepb.Grid
+	if err = proto.Unmarshal(buf, &g); err != nil {
+		return fmt.Errorf("proto.Unmarshal: %w", err)
+	}
+
+	// TODO(chases2): From summarizer: filterGrid()
```

Comment:
IMO this PR is fine but not particularly useful. 

This just demonstrates using the proto and zip packages, which seems unlikely to cause a problem.

The more interesting piece here is to inflate the grid and then deflate it.

AKA convert it into `[]InflatedGrid` and back

https://github.com/GoogleCloudPlatform/testgrid/blob/90494ebb6318b4a135aa950c1d70d087d74a81ab/pkg/updater/inflate.go#L32-L37

One idea might be to call `InflateDropAppend` with a noop column reader:

https://github.com/GoogleCloudPlatform/testgrid/blob/90494ebb6318b4a135aa950c1d70d087d74a81ab/pkg/updater/updater.go#L549

Reply:
I don't think we'll need to inflate/deflate the grid in order to filter it. The summarizer doesn't.

We will need to inflate/deflate to drop columns later, to determine alerting and to drop empty columns after filtering, but I intend to do that in a later PR.